pkg/kms/api.go

```diff
@@ -17,12 +17,46 @@ import (
 // KeyManager manages keys and their storage for the aries framework
 type KeyManager interface {
 	// Create a new key/keyset/key handle for the type kt
+	// Returns:
+	//  - keyID of the handle
+	//  - handle instance (to private key)
+	//  - error if failure
 	Create(kt KeyType) (string, interface{}, error)
 	// Get key handle for the given keyID
+	// Returns:
+	//  - handle instance (to private key)
+	//  - error if failure
 	Get(keyID string) (interface{}, error)
 	// Rotate a key referenced by keyID and return a new handle of a keyset including old key and
 	// new key with type kt. It also returns the updated keyID as the first return value
+	// Returns:
+	//  - new KeyID // TODO remove creation of new keyID from Rotate() - #1837
+	//  - handle instance (to private key)
+	//  - error if failure
 	Rotate(kt KeyType, keyID string) (string, interface{}, error)
+	// ExportPubKeyBytes will fetch a key referenced by id then gets its public key in raw bytes and returns it.
+	// The key must be an asymmetric key.
+	// Returns:
+	//  - marshalled public key []byte
+	//  - error if it fails to export the public key bytes
+	ExportPubKeyBytes(keyID string) ([]byte, error)
```

Comment:
This is a really important addition - to add `ExportPubKeyBytes()` and `PubKeyBytesToHandle()`. Having these functions in the API, we won't need to make type assertion to `LocalKMS`.